### ISSUE TITLE
SC1036 parse error on valid let statement

### ISSUE THREAD

**Issue by smeier** [koalaman/shellcheck#867]
#### Bug
- Rule: SC1036: '(' is invalid here. Did you forget to escape it?
- Version: online
- [ X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
let x=3+5
echo $x

```

#### Here's what shellcheck currently says:

Line 2:
let x=(3+5)*2
    ^-- SC2034: x appears unused. Verify it or export it.
      ^-- SC1036: '(' is invalid here. Did you forget to escape it?
      ^-- SC1088: Parsing stopped here. Invalid use of parentheses?

#### Here's what I wanted or expected to see:

No issues detected!

**Comment by kurahaupo** [comment]
Even though that syntax is accepted by Bash version 4.3, Shellcheck should
NOT say "No issues detected!"

Firstly, this syntax is specific to Bash (for a long time -- since version
2, apparently). For Bourne, ksh, mksh & others, Shellcheck is correct:
brackets are indeed not allowed there. Regardless, it's always been
recommended that the parameters to "let" be quoted if they contain shell
meta-characters, the same as any normal command.

Secondly, even for current Bash, that enhancement does not extend to
disabling globbing; before running your example, type this:

    touch 'x=(3+5)+22'

and then try your unquoted example:

    let x=(3+5)*2
    echo $x

which prints "30" instead of "16".

So you think you're safe because you don't have any "weird" filenames lying
around?

Then try this

    rm x=*
    shopt -s nullglob

and then try your unquoted example:

    let x=(3+5)*2

which now says

    bash: let: expression expected

(and doesn't change the value of x)

Even if you get around both of those, consider how much more slowly your
script is going to run if every arithmetic operation results in an attempted
(but failing) glob expansion.

Thirdly, please use $((...)) or ((...)) instead of let "..."

**Comment by smeier** [comment]
OK. So now instead of:

`No issues detected!`

I'd really  expect to see:

`please use $((...)) or ((...)) instead of let "..."`
